examples/src/bin/gio_task.rs

```diff
@@ -0,0 +1,192 @@
+    /// Task result and perform some operation with it.
+    #[no_mangle]
+    pub unsafe extern "C" fn get_file_size_async(
+        this: *mut FileSize,
+        cancellable: *mut gio::ffi::GCancellable,
+        callback: gio::ffi::GAsyncReadyCallback,
+        user_data: glib::ffi::gpointer,
+    ) {
+        let cancellable = gio::Cancellable::from_glib_none(cancellable);
+        let closure = move |result: &gio::AsyncResult, source_object: Option<&glib::Object>| {
+            let result: *mut gio::ffi::GAsyncResult = result.to_glib_none().0;
+            let source_object: *mut glib::object::GObject = source_object.to_glib_none().0;
+            callback.unwrap()(source_object, result, user_data)
+        };
+
+        let source_object = &super::FileSize::from_glib_none(this);
+        let task = gio::Task::new(
+            Some(&source_object.clone().upcast::<glib::Object>()),
+            Some(&cancellable),
+            closure,
+        );
+
+        glib::MainContext::default().spawn_local(async move {
+            let size = gio::File::new_for_path("Cargo.toml")
+                .query_info_async_future("*", gio::FileQueryInfoFlags::NONE, glib::PRIORITY_DEFAULT)
+                .await
+                .unwrap()
+                .get_size();
+
+            let source_object = task
+                .upcast_ref::<gio::AsyncResult>()
+                .get_source_object()
+                .unwrap();
+
+            let source_object = imp::FileSize::from_instance(
+                &source_object.downcast_ref::<super::FileSize>().unwrap(),
+            );
+
+            source_object.size.replace(Some(size));
+            task.return_value(&size.to_value());
+        });
+    }
+
+    /// # Safety
+    ///
+    /// This method will typically be invoked in the callback passed to get_file_size_async in order
+    /// to propagate the Task result.
+    #[no_mangle]
+    pub unsafe extern "C" fn get_file_size_finish(
+        _this: *mut FileSize,
+        result: *mut gio::ffi::GAsyncResult,
+        _error: *mut *mut glib::ffi::GError,
+    ) -> i64 {
+        gio::AsyncResult::from_glib_none(result)
+            .downcast_ref::<gio::Task>()
+            .unwrap()
+            .propagate_value()
+            .unwrap()
+            .get::<i64>()
+            .unwrap()
+            .unwrap()
+    }
+
+    /// # Safety
+    ///
+    /// Simple getter
+    #[no_mangle]
+    pub unsafe extern "C" fn get_retrieved_size(this: *mut FileSize) -> i64 {
+        let simple_object = super::FileSize::from_glib_none(this);
+        let simple_object =
+            imp::FileSize::from_instance(&simple_object.downcast_ref::<super::FileSize>().unwrap());
+        let x = *simple_object.size.borrow();
+        x.unwrap()
+    }
+}
+
+glib::wrapper! {
+    pub struct FileSize(ObjectSubclass<imp::FileSize>);
+}
+
+impl FileSize {
+    pub fn new() -> Self {
+        glib::Object::new(&[]).expect("Failed to create FileSize")
+    }
```

Comment:
I think for consistency the proper API from the `extern "C"` functions above should be available here too. With the same scheme as in the bindings (but not necessarily calling the FFI functions).